art/attacks/evasion/laser_attack/algorithms.py

```diff
@@ -0,0 +1,74 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Optional, Tuple
+
+import numpy as np
+from art.attacks.evasion.laser_attack.utils import (AdversarialObject,
+                                                    AdvObjectGenerator,
+                                                    DebugInfo, ImageGenerator)
+
+
+# https://openaccess.thecvf.com/content/CVPR2021/papers/Duan_Adversarial_Laser_Beam_Effective_Physical-World_Attack_to_DNNs_in_a_CVPR_2021_paper.pdf
+# Algorithm 1
+def greedy_search(
+    image: np.ndarray,
+    estimator,
+    iterations: int,
+    actual_class: int,
+    actual_class_confidence: float,
+    adv_object_generator: AdvObjectGenerator,
+    image_generator: ImageGenerator,
+    debug: Optional[DebugInfo] = None,
+) -> Tuple[Optional[AdversarialObject], Optional[int]]:
+    """
+    Greedy search algorithm used to generate parameters
+    of an adversarial object that added to the :image
+    will mislead the neural network.
```

Comment:
Use line length=120:
```suggestion
    Greedy search algorithm used to generate parameters of an adversarial object that added to the :image will mislead
    the neural network.
```